platform-hub-api/app/controllers/concerns/authentication.rb

```diff
@@ -0,0 +1,50 @@
+module Authentication
+  extend ActiveSupport::Concern
+
+  include ActionController::HttpAuthentication::Token
+
+  def require_authentication
+    head :unauthorized unless authenticated?
+  end
+
+  def authenticated?
+    current_user.present?
+  end
+
+  def current_user
+    token, _ = token_and_options(request)
+
+    return nil if token.blank?
+
+    @current_user ||= create_or_fetch_user(token)
+  end
+
+  private
+
+  def create_or_fetch_user token
+    # Important assumptions about the token:
+    # - it is already verified by the auth proxy in front
+    # - expiry has already been checked by the auth proxy in front (TODO: perhaps recheck here)
+    begin
+      payload, _ = JWT.decode(token, nil, false)
+    rescue => e
+      logger.error "Failed to parse JWT token: #{token}. Error: #{e.message}"
+      return nil
+    end
+
+    return nil if payload.blank? || payload['sub'].blank?
+
+    id = payload['sub']
+
+    user = User.find_by_id(id)
+
+    if user.blank?
+      user = NewUserService.new.create payload
```

Comment:
I'm not a fan of static methods since they become one instance for the _whole_ runtime, instead of isolating effects to as small an area as possible (albeit this particular service is stateless and behaves in a highly functional way). Given this is happening within a request/response cycle, it's fairly common to have service objects be instantiated and used just the cycle.

In the future, if this becomes a perf problem, we can have a higher level area in which to instantiate singleton-esque instances of various services and then inject them we where need (makes it nicer to test).